kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -294,12 +300,9 @@ private boolean rotateOnTime(String encodedPartition, Long recordTimestamp, long
       return false;
     }
     // rotateIntervalMs > 0 implies timestampExtractor != null
-    boolean periodicRotation = rotateIntervalMs > 0
-        && timestampExtractor != null
-        && (
-        recordTimestamp - baseRecordTimestamp >= rotateIntervalMs
-            || !encodedPartition.equals(currentEncodedPartition)
-    );
+    boolean periodicRotation = rotateIntervalMs > 0 && timestampExtractor != null;
```

Comment:
again `strictPartitioning` is counter-intuitive. It'll be better to signal the new feature as `true` and the old functionality as `false` for the append feature. 

In any case, splitting this boolean expression is correct. We might also need comments though re: what every variable is meant to mean